core/src/main/java/org/testcontainers/utility/DefaultImageNameSubstitutor.java

```diff
@@ -0,0 +1,47 @@
+package org.testcontainers.utility;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * TODO: Javadocs
+ */
+@Slf4j
+public class DefaultImageNameSubstitutor extends ImageNameSubstitutor {
```

Comment:
I've made this as a default out-of-the-box substitutor that delegates to both the config file settings _and_ a simple 'prefix' substitutor which can apply a common prefix to all image names.

I suspect this might be enough for 80% of situations where people need a substitutor, which is why I think it's worth including it by default.

Reply:
FTR for others looking at this PR:
this comment is outdated, see #3413 for the prefixing substitutor